packages/backend/src/formatter.ts

```diff
@@ -0,0 +1,56 @@
+import { Explore, fieldId as getFieldId, getFields, ResultRow } from 'common';
+
+function formatValue(format: string, value: any): any {
+    switch (format) {
+        case 'km':
+        case 'mi':
+            return `${value} ${format}`;
+        case 'usd':
+            return `$${value}`;
+        case 'gbp':
+            return `${value}£`;
+        case 'percent':
+            return `${parseFloat(value) * 100} %`;
```

Comment:
What happens if this fails? We return `NaN`? That feels ok, we could optionally add protection during the `translator.ts` step. But let's see how we go.



The return type could be tightened up:

`formatValue<T>(format: string, value: T): string | T`